appscale/tools/remote_helper.py

```diff
@@ -122,7 +121,24 @@ def start_head_node(cls, options, my_id, node_layout):
       options.verbose)
 
     if options.infrastructure:
-      instance_id, public_ip, private_ip = cls.spawn_node_in_cloud(options)
+      agent = InfrastructureAgentFactory.create_agent(options.infrastructure)
+
+      params = agent.get_params_from_args(options)
+      public_ips, private_ips, instance_ids = agent.describe_instances(params)
+      try:
+        login_ip = LocalState.get_login_host(options.keyname)
+      except (IOError, BadConfigurationException):
+        # We don't have the location file, we need to start instances.
+        login_ip = None
+
+      if login_ip in public_ips:
```

Comment:
I would be a little more defensive about this rather than relying on `describe_instances` to never return `None` in the list of `public_ips`.